api/src/controllers/users.js

```diff
@@ -225,4 +225,34 @@ module.exports = {
       }))
       .catch(err => serverUtils.error(err, req, res));
   },
+
+  v2: {
+    create: async (req, res) => {
+      try {
+        await auth.check(req, 'can_create_users');
+        const logId = await bulkUploadLog.createLog(req, 'user');
+        let users;
+        let ignoredUsers;
+
+        if (typeof req.body === 'string') {
+          const parsedCsv = await usersService.parseCsv(req.body, logId);
+          users = parsedCsv.users;
+          ignoredUsers = parsedCsv.ignoredUsers;
+        } else {
+          users = req.body;
+        }
```

Comment:
Please move this logic elsewhere. It'll leave you with something like this, which will make the larger function easier to reason about:

```
        const users = await loadUsers(req.body);
```